maestrowf/interfaces/script/_flux/flux0_49_0.py

```diff
@@ -46,6 +122,125 @@ def get_flux_urgency(cls, urgency) -> int:
+        unflattened_batch_args = {
+            arg_type: {}
+            for arg_type in cls.addtl_alloc_arg_types()
+        }
+
+        for arg_key, arg_values in args_dict.items():
+            arg_type = cls.addtl_alloc_arg_type_map(arg_key)
+            # print(f"normalize_additional_args: {arg_key=} -> {arg_type=}")
+            if arg_type is None:
+                # We only restrict for allocations, so defer logging/rejection
+                # of types not handled by jobspec builder
+                arg_type = arg_key
+
+            if isinstance(arg_values, dict):
+                unflattened_batch_arg = {arg_type: unflatten_dotpath_dict(arg_values)}
+                # unflattened_batch_args[arg_key] = unflatten_dotpath_dict(arg_values)
+            else:
+                unflattened_batch_arg = {arg_type: arg_values}
+                # unflattened_batch_args[arg_key] = arg_values
+
+            # Update to ensure we don't clobber prior values
+            # print(f"pre-update  unflattened_batch_args: {unflattened_batch_args=}")
+            unflattened_batch_args = update_recursive(unflattened_batch_args,
+                                                      unflattened_batch_arg)
+        #     print(f"post-update unflattened_batch_args: {unflattened_batch_args=}")
+
+        # print(f"normalized_additional_args: {unflattened_batch_args=}")
+        return unflattened_batch_args
+
+    @classmethod
+    def pack_addtl_batch_args(cls, args_dict):
+        """
+        Normalize the allocation args and pack up into the interface specific
+        groups that have assocated jobspec methods, e.g. conf, setattr, setopt.
+        Ensure arg formats match their end point's types.  For current flux
+        versions as of 0.78.0: setattr and setopt require dicts be flattened
+        into lists of dotpath encoded strings and values, while conf requires
+        a completely unflattened dict.
+
+        :param args_dict: Dict with keys normalized to cls.addtl_alloc_arg_types
+        :return: dictionary of allocation arg groups to attach to jobspecs
+        """
+        # Normalize None values common in flag inputs for use in python api
+        # see https://github.com/flux-framework/flux-core/blob/a3860d4dea5b5a17c473cff4385276e882275252/src/bindings/python/flux/cli/base.py#L734
+        # NOTE: only doing this in alloc; LAUNCHER cli passes through
+        #       to flux cli (None values are omittied, e.g.
+        #       {o: fastload: None} renders to -o fastload
+        #       Python api doesn't appear to have default value handling?
+        packed_batch_args = {
+            arg_type: {}
+            for arg_type in cls.addtl_alloc_arg_types()
+        }
+
+        dotpath_format = ["attributes", "shell_options"]
+
+        for arg_key, arg_values in args_dict.items():
+            coerced_vals = coerce_dict_values(arg_values,
+                                              lambda x: 1 if x is None else x)
```

Comment:
kinda unbelievable that this is required and flux python doesn't handle this for us